### PR TITLE
Remove coveralls

### PR DESCRIPTION
**Pull request by benSlaughter**
The Jenkins tests were giving a false positive because of coveralls.
I have removed coveralls, fixed some rake and rubocop issues.

- Removed coveralls
- Updated rubocop to require at minimum 0.38
- Updated gemspec to include at least rake 12.0
- Fixed some minor rubocop issues
- Bumped to version 13.2.0 as there are no breaking changes to the functionality here

### CODE REVIEW

**Comment by benSlaughter** [comment]
test this please

File: lib/bwapi/client/oauth.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ def oauth_token(opts = {})
 
         oauth_request opts
       end
-      alias_method :login, :oauth_token
+      alias login oauth_token
```
Comment (SirCN) [inline lib/bwapi/client/oauth.rb]: Is this change intended?
IIRC alias_method => alias changes the scope?

Comment (benSlaughter) [inline lib/bwapi/client/oauth.rb]: This was something that was picked up by rubocop.
[Here](https://github.com/bbatsov/ruby-style-guide#alias-method-lexically) is the entry in the style guide.
It looks like it is to have the code as clear as possible as to what is happening. Although the part about inheritance is interesting. Makes sense though.

Comment (SirCN) [inline lib/bwapi/client/oauth.rb]: Make sense, thanks for the explanation!
